go/vt/vttablet/tabletserver/txthrottler/tx_throttler.go

```diff
@@ -266,14 +272,34 @@ func (t *TxThrottler) Close() {
 // It returns true if the transaction should not proceed (the caller
 // should back off). Throttle requires that Open() was previously called
 // successfully.
-func (t *TxThrottler) Throttle(priority int) (result bool) {
+func (t *TxThrottler) Throttle(priority int, planType *planbuilder.PlanType) (result bool) {
 	if !t.config.enabled {
 		return false
 	}
 
-	// Throttle according to both what the throttle state says, and the priority. Workloads with lower priority value
+	// This blocks handles deciding whether to apply the throttle or not. In
+	// summary, it will apply the throttling logic in any of these cases:
+	// - The plan type is nil (as during TabletServer.begin()) AND throttling auto-commit statements is disabled;
+	// - The plan type is for a write (INSERT/UPDATE/DELETE/LOAD) AND throttling auto-commit statements is enabled;
+	applyThrottling := false
+	if planType == nil && !t.config.autoCommit {
+		applyThrottling = true
+	}
+	if planType != nil && t.config.autoCommit {
+		switch *planType {
+		case planbuilder.PlanInsert, planbuilder.PlanInsertMessage, planbuilder.PlanUpdate, planbuilder.PlanUpdateLimit, planbuilder.PlanDeleteLimit, planbuilder.PlanDelete, planbuilder.PlanLoad:
+			applyThrottling = true
+			break
```

Comment:
this check looks very brittle to me. If we look at query_executor all the queries go through 
`qre.tsv.te.txPool.Begin` call. The place we can throttle is in `txpool` -> `begin` that way it would not matter how we reached this method.

Reply:
The reason throttling was done at transaction creation is that the transaction does not open for a long time and if they do get open should be allowed to do the commit.
With this change, the connections will get blocked to eventually get throttled and rolled back by transaction killer.

Reply:
I am afraid I am not following how this fits into my comment/question - perhaps I explained poorly.
What I meant to ask you here is whether - after moving the `TxThrottler.Throttle()` call from its current location in `TabletServer.begin()` to `TxPool.Begin()`, would it be acceptable for the `Throttle()` method to still check the plan type as in the code block you started this thread on - that is, to consider throttling if the plan type is nil or is insert/update/delete/load.

Reply:
Let me move that discussion to the Feature Request. For the plan type that will be available after the query plan is built and would not be available at `begin` time. So, this can still lead to a miss where any new plan types related to dml if added can get missed.